### PR TITLE
Fix/tca 678/rest put request

### PR DESCRIPTION
**Pull request by zagovorichev**
Pull request summary
 
Related to: https://oat-sa.atlassian.net/browse/TCA-678
 
REST API doesn't allow us to use a data from the body, that required with PUT action.
 
#### Steps to reproduce  (for bugs)
 - send PUT with the data that has to be updated in the resource
 - nothing changed
  
#### How to test
 
```
curl --location --request PUT 'http://tao-act.loc/taoLti/RestService/index?uri=http%3A%2F%2Fsample%2Fact.rdf%23i5ef44f33c222b694e6664782566e0038' \
--header 'Accept: application/json' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--header 'Authorization: Basic ###' \
--data-raw '{
    "oauth_consumer_key": "valfromput2",
    "oauth-key": "valfromput2_oauth",
    "label": "labelfromput2",
    "oauth-secret": "secretfromput2",
    "oauth-callback-url": "callbackfromput2"
}'
```

**uri** - ID of the LTI consumer
 
or use any other tools like a Storm's REST tool or Postman.

### CODE REVIEW

**Comment by zagovorichev** [comment]
@augustas I've added accepted header checker, but what do you think: do I need to add xml parser here? 
(according to this commentary: https://github.com/oat-sa/tao-core/pull/2574#discussion_r446102487)

File: actions/class.CommonRestModule.php [hunk 1]
```diff
@@ -221,21 +221,43 @@ protected function put($uri)
...
+
     /**
      * Returns all parameters that are URIs or Aliased with values
      *
      * @throws \common_exception_RestApi If a mandatory parameter is not found
      *
      * @return array
      */
-    protected function getParameters()
+    protected function getParameters(): array
```
Comment (SashaTAO) [inline actions/class.CommonRestModule.php:255]: If we add return type declaration here we have to add it in classes which extend this one.

File: actions/class.CommonRestModule.php [hunk 2]
```diff
@@ -221,21 +221,43 @@ protected function put($uri)
...
+            $parameters = [];
+        }
+
+        return $parameters;
+    }
+
+    private function getBodyData(): array
+    {
+        $data = $this->getPsrRequest()->getBody()->getContents();
+        if ($data && $this->hasHeader('Accept') && $this->getHeader('Accept') === 'application/json') {
```
Comment (SashaTAO) [inline actions/class.CommonRestModule.php:236]: `$this->getHeader('Accept')` returns an array and condition is always `false`.

File: actions/class.CommonRestModule.php (hunk 1 above)
Comment (zagovorichev) [reply actions/class.CommonRestModule.php]: done

File: actions/class.CommonRestModule.php (hunk 2 above)
Comment (zagovorichev) [reply actions/class.CommonRestModule.php]: fixed

**Pull request merged**